linker/shared/src/main/scala/org/scalajs/linker/analyzer/Infos.scala

```diff
@@ -684,6 +690,20 @@ object Infos {
           // Capture values are in the enclosing scope; not the scope of the closure
           captureValues.foreach(traverse(_))
 
+        // Do not call super.traverse(), as we must follow a single branch
+        case LinkTimeIf(cond, thenp, elsep) =>
+          builder.markNeedsDesugaring()
+          traverse(cond)
+          coreSpec.linkTimeProperties.tryEvalLinkTimeBooleanExpr(cond) match {
+            case Some(result) =>
+              if (result)
+                traverse(thenp)
+              else
+                traverse(elsep)
+            case None =>
+              () // ignore; the `traverse(cond)` will issue at least one linking error
```

Comment:
This comment doesn't feel like it is correct.

`traverse(cond)` will for sure not issue a linking error here (this is the info loader).

Looking at the entire linking pipeline, it seems that we are only checking `cond` when IR checking is enabled and in the desugarer when we actually desugar the node. IMO this is OK, maybe:

- Adjust the comment here.
- Add a text in the assertion error in desguar to tell users to enable IR checking before reporting a bug.